clang/test/CodeGenSYCL/kernel-by-reference.cpp

```diff
@@ -15,15 +15,15 @@ int simple_add(int i) {
 int main() {
   queue q;
 #if defined(SYCL2020)
-  // expected-warning@Inputs/sycl.hpp:301 {{Passing kernel functions by value is deprecated in SYCL 2020}}
+  // expected-warning@Inputs/sycl.hpp:311 {{Passing kernel functions by value is deprecated in SYCL 2020}}
```

Comment:
This test fails due to wrong line number reference.
I think exact like numbers are not that important for all updated tests. I suggest we replace numbers with matching pattern to avoid test changes each time we modify `Inputs/sycl.hpp` if these changes are just line number in the header.

```suggestion
  // expected-warning@Inputs/sycl.hpp:{{[0-9]+}} {{Passing kernel functions by value is deprecated in SYCL 2020}}
```

Reply:
I'd suggest instead putting a marker in the sycl.hpp header.   See the part below from the documentation.

Essentially, we put in the header on the offending line:  // #SomethingUniqueForThisKernelFunction

Then, anytime we see it,  we do:
// expected-warning@#SomethingUniqueForThisKernelFunction {{Passing...blah.blah.blah}}


As an alternative to specifying a fixed line number, the location of a diagnostic can instead be indicated by a marker of the form "#<marker>". Markers are specified by including them in a comment, and then referenced by appending the marker to the diagnostic with "@#<marker>":
#warning some text  // #1
// expected-warning@#1 {{some text}}

The name of a marker used in a directive must be unique within the compilation.

Reply:
> I'd suggest instead putting a marker in the sycl.hpp header. See the part below from the documentation.
...
> Markers are specified by including them in a comment, and then referenced by appending the marker to the diagnostic with "@#":
> #warning some text // #1
> // expected-warning@#1 {{some text}}
> 
> The name of a marker used in a directive must be unique within the compilation.

This is neat!  I did not know this before.

Reply:
Thanks for the tip! Yea all the tests fail because I changed the header in last commit. I'll make this change

Reply:
The following is not working. I'm not sure what I'm doing wrong.

```
-ATTR_SYCL_KERNEL void kernel_single_task_2017(KernelType kernelFunc) {
+ATTR_SYCL_KERNEL void kernel_single_task_2017(KernelType kernelFunc) { // #KernelSingleTask2017

Test
-  // expected-warning@Inputs/sycl.hpp:311 {{Passing kernel functions by value is deprecated in SYCL 2020}}
+  // expected-warning@Inputs/sycl.hpp@#KernelSingleTask2017 {{Passing kernel functions by value is deprecated in SYCL 2020}}
```
I get this error - 

```
error: 'error' diagnostics seen but not expected:
  File xyz/kernel-by-reference.cpp Line 18: missing or invalid line number following '@' in expected string
```

Reply:
I tried replacing the marker with a numerical one but run into the same issue. I assume its something to do with the fact that the marker is in a different file. Not sure if I got the syntax right for specifying file+marker

Reply:
I figured it out :)  I should have read the error message better.  That expected-warning doesn't need the filename, and shouldn't.

Reply:
Done! Thanks!